### ISSUE TITLE
[Rule change] avoid-late-keyword to allow late for AnimationControllers

### ISSUE THREAD

**Issue by illia-romanenko** [dart-code-checker/dart-code-metrics#1007]
**What rule do you want to change?**
avoid-late-keyword

**How will the change be implemented? (New option, new default behavior, etc.)?**
new behavior
**Please provide some example code that this change will affect:**

It looks like using late is pretty idiomatic for AnimationController: https://api.flutter.dev/flutter/animation/AnimationController-class.html

```dart
class _FooState extends State<Foo> with SingleTickerProviderStateMixin {
  late AnimationController _controller;

  @override
  void initState() {
    super.initState();
    _controller = AnimationController(
      vsync: this, // the SingleTickerProviderStateMixin
      duration: widget.duration,
    );
  }
```

**What does the rule currently do for this code?**
Flags
**What will the rule do after it's changed?**
Don't flag late with AnimationControllers

**Are you willing to submit a pull request to implement this change?**
I'm not sure yet.

**Comment by incendial** [comment]
@illia-romanenko so, on one hand I understand the idea, but on the other hand this is general dart rule, hardcoding ignores just for AnimationController looks not that flexible as I'd prefer. Any ideas how to make this configurable from the user standpoint?

**Comment by solid-software** [comment]
If we can pass parameters to the rule and include classes to ignore that would do it too.

**Comment by russellminnich** [comment]
Would initializing the value in initState() and cleaning up in dispose() satisfy as initialization?  That's an idiomatic way to use TextEditingController in stateful widgets.

**Comment by incendial** [comment]
The initial idea of the rule is to completely disallow `late`, so even this is an idiomatic way - if you use `late`, maybe you should not enable the rule then?

**Comment by incendial** [comment]
Available in 5.5.0 🚀
